packages/cli-e2e/__tests__/vue.specs.ts

```diff
@@ -0,0 +1,113 @@
+import {spawn, spawnSync} from 'child_process';
+
+import type {ChildProcessWithoutNullStreams} from 'child_process';
+
+import {CLI_EXEC_PATH, killCliProcessFamily} from '../utils/cli';
+import {closeAllPages, getBrowser} from '../utils/browser';
+import {Browser} from 'puppeteer-core';
+import {loginWithOffice} from '../utils/login';
+
+describe('ui', () => {
+  describe('create:vue', () => {
+    let browser: Browser;
+    const cliProcesses: ChildProcessWithoutNullStreams[] = [];
+    // TODO: CDX-90: Assign a dynamic port for the search token server on all ui projects
+    const clientPort = '8080';
+    const projectName = 'vue-project';
+    const searchPageEndpoint = `http://localhost:${clientPort}`;
+    const tokenProxyEndpoint = `http://localhost:${clientPort}/token`;
+    let waitForProjectToStartTimeout: NodeJS.Timeout;
+
+    beforeAll(async () => {
+      browser = await getBrowser();
+
+      await loginWithOffice(browser, cliProcesses);
+
+      const buildProcess = spawnSync(CLI_EXEC_PATH, [
+        'ui:create:vue',
+        projectName,
+      ]);
+
+      expect(buildProcess.status).toEqual(0);
+
+      const waitForProjectToStart = new Promise<void>((resolve) => {
+        const startServerProcess = spawn('npm', ['run', 'start'], {
+          cwd: projectName,
+          detached: true,
```

Comment:
So I finally manage to prevent process leaking.
Every time I spawn a process that will later create its own family of processes, I am using the `detach` option.

> On non-Windows platforms, if `options.detached` is set to true, the child process will be made the **leader** of a new process group and session
[reference](https://nodejs.org/api/child_process.html#child_process_options_detached)

That way, the entire process family can be killed using the leader process id.
It is equivalent to run the following command `kill -TERM -- -<leader-process>`. 
Notice the negative sign before the process id